src/doc_table.c

```diff
@@ -316,8 +325,8 @@ void DocTable_AOFRewrite(DocTable *t, const char *indexName, RedisModuleIO *aof)
       byteOffsetsLen = offsetsBuf.offset;
     }
 
-    RedisModule_EmitAOF(aof, "FT.DTADD", "cclsbb", indexName, dmd->key, dmd->flags, ss, payload,
-                        payloadLen, byteOffsets, byteOffsetsLen);
+    RedisModule_EmitAOF(aof, "FT.DTADD", "cblsbb", indexName, dmd->keyPtr, sdslen(dmd->keyPtr),
```

Comment:
That's binary safe, right? since AOF uses RESP format, this would cause a bulk record to be written.

Reply:
I believe so. We're already copying the byte offsets here anyway, so I don't believe there is any added risk.